src/main/java/com/example/controllers/TransferController.java

```diff
@@ -0,0 +1,145 @@
+
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+
+		Transfer newTransfer = new Transfer();
+		newTransfer.setFee(5.00);
+
+		if (!model.containsAttribute("transfer")) {
+			model.addAttribute("transfer", newTransfer);
+		}
+		model.addAttribute("cashAccounts", cashAccounts);
+		model.addAttribute("account", account);
+
+		response.addCookie(new Cookie("accountType", AccountType.PERSONAL));
+
+		return "newTransfer";
+	}
+
+	@RequestMapping(method = RequestMethod.POST)
+	public String transfer(@Valid @ModelAttribute final Transfer transfer, final BindingResult bindingResult, final Model model,
+			final Principal principal, @CookieValue(value = "accountType", defaultValue = AccountType.PERSONAL) final String accountType,
+			final HttpSession session, final HttpServletResponse response) {
+
+		if (bindingResult.hasErrors()) {
+			return newTransferForm(model, principal, response);
+		}
+		else {
+			return AccountType.PERSONAL.equals(accountType) ? transferCheck(transfer, model, session, principal)
+					: transferConfirmation(transfer, model, principal, accountType);
+		}
+	}
+
+	private String transferCheck(final Transfer transfer, final Model model, final HttpSession session, final Principal principal) {
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+		model.addAttribute("account", account);
+		model.addAttribute("transferbean", transfer);
+		model.addAttribute("operationConfirm", new OperationConfirm());
+		session.setAttribute(PENDING_TRANSFER, transfer);
+
+		return "transferCheck";
+	}
+
+	private String transferConfirmation(final Transfer transfer, final Model model, final Principal principal, final String accountType) {
+
+		List<CashAccount> cashAccounts = cashaccountDao.findCashAccountsByUsername(principal.getName());
+		Account account = accountDao.findUsersByUsername(principal.getName()).get(0);
+		double aux = transfer.getAmount();
+
+		if (aux == 0.0) {
+
+			model.addAttribute("transfer", transfer);
+			model.addAttribute("cashAccounts", cashAccounts);
+			model.addAttribute("account", account);
+			model.addAttribute("error", true);
+
+			return "newTransfer";
+		}
+
+		transfer.setUsername(principal.getName());
+		transfer.setDate(new Date());
+
+		double amount = transfer.getAmount();
+		transfer.setAmount(InsecureBankUtils.round(amount, 2));
+
+		double feeAmount = transfer.getAmount() * transfer.getFee() / 100.0;
+		transfer.setFee(InsecureBankUtils.round(feeAmount, 2));
+
+		transfersFacade.createNewTransfer(transfer);
+
+		model.addAttribute("transferbean", transfer);
+		model.addAttribute("account", account);
+		model.addAttribute("accountType", accountType);
+
+		return "transferConfirmation";
+	}
+
+	@RequestMapping(value = "/confirm", method = RequestMethod.POST)
+	public String transferCheck(final OperationConfirm operationConfirm, final BindingResult bindingResult, final Model model,
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 673013533ee6aa1526c81e1548e79739 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Cross-Site Request Forgery (CSRF)
_High CWE-352_
The web application controller modifies data or state and is unprotected from cross-site request forgery.

Controllers that execute actions or tasks may be accessed to perform activities without the user's knowledge or authorization. These attacks are difficult to distinguish from legitimate requests.

### How to fix
Generate a sufficiently large, non-guessable random token value, and scope it to the current user's session. The CSRF token should not be accepted for any other user across the application.

One way to add a non-predictable parameter is to use a secure hash function, such as SHA-256, to hash the user's session ID. However, many application frameworks provide some built-in support to enable CSRF protection, and it is often recommended to leverage these first whenever available.

A classic pattern for CSRF protection is to generate and store a token in the user's session, and then include the CSRF token in a hidden field within all forms. The server will validate the CSRF token (sometimes referred to as a "form-authenticator") to ensure that the request is legitimate for the current user. If the hidden field and server-side tokens match, the request can be processed; otherwise, the application issues an error indicating the request is invalid. Since an attacker can't access this token for other users, it will not be possible to generate a valid request for another user. Alternative patterns rely on matching a token within the request data against a cookie or custom HTTP header (e.g., "X-XSRF-TOKEN" in AngularJS).

The "SameSite" cookie attribute can be used to prevent a cookie from being included in requests initiated from another site, and therefore reduces the risk of a successful CSRF attack. This attribute supports three different values:

* None - cookies for the site are always sent with requests. This is the default behavior in older browsers and provides no protection against CSRF. Note that cookies without the "Secure" attribute will be blocked from setting "SameSite=None" by browsers.
* Lax - cookies for the site will normally not be sent on cross-site requests, but will be included for top-level navigation (e.g., when a user clicks on a link from a different site). This is the default behavior in recent versions of some modern browsers, such as Chrome and Edge, and provides limited protection against some CSRF attacks.
* Strict -  cookies for the site will only be included with same-site requests, and never included with cross-site requests. This is the most secure setting of the three but may have some negative impact on user experience, e.g., cross-site navigation to a site where the user has already logged in will still appear to the user as if they were not authenticated. 

It's worth noting that the definition of same-site is less restrictive than the definition of cross-origin, e.g., requests from other sub-domains are considered same-site although they are from a different origin. This means that even in browsers that fully support the "SameSite" cookie attribute, an attacker may still circumvent this protection if they can take over a subdomain or inject HTML or JavaScript into any subdomain of the same site. Therefore, relying on the "SameSite" attribute is not recommended as the sole defense against cross-site request forgery, it should instead be seen as a defense in depth mechanism that reduces the likelihood of CSRF.